### ISSUE TITLE
`scoped: true`: type leaks, lifecycle hooks don't go off, and state never gets initialized

### ISSUE THREAD

**Issue by ethanniser** [elysiajs/elysia#127]
I'm trying to used `scoped: true` and im running into some strange behavior

1. `use`ing an Elysia instace with `scoped: true`, doesn't remove the types from that instance, even though they shouldn’t be accessible

2. `store` values in Elysia instaces with `scoped: true` seem to be wrongfully initiazed as undefined

3. `scoped` lifecycle hooks don't seem to be working

```ts
import { Elysia } from "elysia";

const scoped = new Elysia({
  name: "scoped",
  scoped: true,
})
  .state("count", 0)
  .get("/scoped", ({ store }) => {
    console.log("store:", store);
    store.count += 1;
    return store.count;
  })
  .onRequest(() => {
    console.log("this should only trigger for /scoped");
  });

const app = new Elysia()
  .use(scoped)
  .get("/", ({ store }) => {
    console.log("store (should be empty):", store);
    console.log("this should not be possible:", store.count);
    // ^^ doesn't type error
    return "Hello World!";
  })
  .onRequest(({ request }) => {
    console.log(`request: ${request.method} ${request.url}`);
  })
  .listen(8080);

console.log(
  `🦊 Elysia is running at http://${app.server?.hostname}:${app.server?.port}`
);
```

```
🦊 Elysia is running at http://localhost:8080

request: GET http://localhost:8080/
store (should be empty): {}
this should not be possible: undefined

request: GET http://localhost:8080/scoped
store: {}

request: GET http://localhost:8080/scoped
store: {
  count: NaN
}

request: GET http://localhost:8080/
store (should be empty): {
  count: NaN
}
this should not be possible: NaN
```

**Comment by SaltyAom** [comment]
Ok, seems like there's some area I haven't cover yet. I'll handle that in 0.7.
As a work around, does wrapping the scope inside a guard hook should fix the problem.

**Comment by SaltyAom** [comment]
This should be fixed in 0.7 #103 

https://elysiajs.com/blog/elysia-07.html#true-encapsulation-scope
